src/Xeno/DOM.hs

```diff
@@ -195,6 +199,21 @@ parse str =
                  -- Pop the stack and return to the parent element.
                  previousParent <- UMV.read v (parent + 1)
                  writeRef parentRef previousParent)
+              (\(PS _ cdata_start cdata_len) -> do
```

Comment:
This looks like a verbatim copy of the "Text" handler, which is good. It means it probably works fine and isn't a concern to merge.